experimental/framework/fluid-static/src/types.ts

```diff
@@ -71,3 +71,70 @@ export interface ContainerSchema {
      */
     dynamicObjectTypes?: LoadableObjectClass<any>[];
 }
+
+/**
+ * Base interface to be implemented to fetch each service's audience. The generic C allows consumers to further
+ * extend the client object with service-specific details about the connecting client, such as device information,
+ * environme
+ */
+export interface IServiceAudience<M extends IMember> {
+    /**
+     * Returns an map of all users currently in the Fluid session where key is the userId and the value is the
+     * member object
+     */
+    getMembers(): Map<string, M>;
+
+    /**
+     * Returns the current active user on this client once they are connected. Otherwise, returns undefined.
+     */
+    getMyself(): M | undefined;
+
+    /**
+     * Returns the current client's details once it is connected. Otherwise, returns undefined.
+     */
+    getMyClient(): IConnectedClient | undefined;
+
+    /**
+     * Collection of event listeners that trigger if the members in the Fluid session change and if a member makes an
+     * edit which updates when the last change was made
+     */
+    on(event: "membersChanged", listener: (members: M[]) => void): this;
+    on(event: "lastEditedChanged", listener: (result: ILastEditedResult<M>) => void): this;
+    off(event: "membersChanged", listener: (members: M[]) => void): this;
+    off(event: "lastEditedChanged", listener: (result: ILastEditedResult<M>) => void): this;
+    once(event: "membersChanged", listener: (members: M[]) => void): this;
+    once(event: "lastEditedChanged", listener: (result: ILastEditedResult<M>) => void): this;
+}
+
+/**
+ * Base interface for providing client information for each connection made to the Fluid session, which will be
+ * different even if it is by the same user, i.e. the clientId will be uniquely generated for each time the user
+ * connects and each indvidiual connection will track its own time that it was last active at. This interface
+ * can be extended to provide additional information specific to each service.
+ */
+export interface IConnectedClient {
+    clientId: string;
+    connectionMode: "write" | "read";
+    timeLastActive?: Date;
+}
+
+/**
+ * Base interface to be implemented to fetch each service's member. The user ID is unique for each individual
+ * user that is connecting to the session. However, one user may be connecting through multiple clients and
+ * the information for each is provided within connectedClients. The list of clients here is sorted based on
+ * descending order from who made the most recent edit. This interface can be extended by each service
+ * to provide additional service-specific user metadata.
+ */
+export interface IMember {
```

Comment:
I'm having a hard time keeping it straight what the difference/relationship between `IConnectedClient` vs `IMember` is (and similarly the API surface like "Myself" vs. "MyClient").  No change recommended for this PR necessarily, but wondering if there's some alternate naming we could consider to help clarify?

Reply:
I believe with the change to remove any last edited state from audience, we can just keep the `getMyself` call and the `IMember` object. There shouldn't be anymore need for an `IConnectedClient` object. Also, perhaps in the future, we could name it "connection"? One member could have multiple `connectionId`s assigned to them based on connecting from different tabs or devices

Reply:
I was thinking we could start calling them "connections" since it represents each unique connection to the session the member makes. I've renamed `IConnectedClient` here to be `IConnection` to avoid confusion also with multiple client-references, i.e. `TinyliciousClient`. Each `IMember` now holds a connections array.
`IConnection` now just holds the id (fetched from the container's clientId) and the mode (read vs write) to allow for any different UI representations based on which mode the client is joined in with, the last edited time is removed

Reply:
Nice, I like that a lot better.